pkg/broker/broker.go

```diff
@@ -185,18 +189,51 @@ func (a AnsibleBroker) Bootstrap() (*BootstrapResponse, error) {
 		a.log.Error("Something went real bad trying to delete batch specs... - %v", err)
 		return nil, err
 	}
+	specs = []*apb.Spec{}
 
-	if specs, imageCount, err = a.registry.LoadSpecs(); err != nil {
-		return nil, err
+	//Load Specs for each registry
+	registryErrors := []error{}
+	for _, r := range a.registry {
+		s, i, err := r.LoadSpecs()
+		if err != nil && r.Fail(err) {
+			a.log.Errorf("registry caused bootstrap failure - %v", err)
+			return nil, err
+		}
+		if err != nil {
+			registryErrors = append(registryErrors, err)
+		}
+		imageCount += i
+		addNameAndIDForSpec(s, r.RegistryName())
+		specs = append(specs, s...)
 	}
-
-	if err := a.dao.BatchSetSpecs(apb.NewSpecManifest(specs)); err != nil {
+	if len(registryErrors) == len(a.registry) {
+		return nil, errors.New("all registries failed on bootstrap")
+	}
+	specManifest := map[string]*apb.Spec{}
+	for _, s := range specs {
+		specManifest[s.ID] = s
+	}
+	if err := a.dao.BatchSetSpecs(specManifest); err != nil {
 		return nil, err
 	}
+	a.log.Debugf("specs -> %v", specs)
 
 	return &BootstrapResponse{SpecCount: len(specs), ImageCount: imageCount}, nil
 }
 
+func addNameAndIDForSpec(specs []*apb.Spec, registryName string) {
```

Comment:
this is an unexported method, so it wont complain. Do you think there needs to be a comment?